File: app/src/main/java/com/onwordiesquire/mobile/marvelapp/presentation/SearchCharacter/SearchView.java
package com.onwordiesquire.mobile.marvelapp.presentation.SearchCharacter;

import com.onwordiesquire.mobile.marvelapp.data.model.CharacterData;
import com.onwordiesquire.mobile.marvelapp.data.model.RecentSearches;
import com.onwordiesquire.mobile.marvelapp.presentation.MvpView;

import java.util.List;

/**
 * Created by michelonwordi on 10/24/16.
 */

public interface SearchView extends MvpView {

    public void displayLastFiveSearches(List<RecentSearches> searches);

    void showError(String message);

    void showEmptyState();

    void showCharacterDetails(CharacterData characterData);

    void showProgressIndicator(boolean value);
}

File: app/src/main/java/com/onwordiesquire/mobile/marvelapp/presentation/SearchCharacter/refactor/SearchView.kt
package com.onwordiesquire.mobile.marvelapp.presentation.SearchCharacter.refactor

import com.onwordiesquire.mobile.marvelapp.data.refactor.data.MarvelCharacter
import com.onwordiesquire.mobile.marvelapp.data.refactor.data.RecentSearches
import com.onwordiesquire.mobile.marvelapp.presentation.MvpView

/**
 * Created by michel.onwordi on 10/07/2017.
 */
interface SearchView : MvpView {
    fun displayLastFiveSearches(searches: List<RecentSearches>)
    fun showError(message: String)
    fun showEmptyState()
    fun showCharacterDetails(marvelCharacter: MarvelCharacter)
    fun showProgress(visible: Boolean)
}